core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -98,6 +103,14 @@ public static <T> T parse(String json, FromJson<T> parser) {
     }
   }
 
+  public static <T, A> T parseWithArg(String json, FromJsonWithArgs<T, A> parser, A args) {
+    try {
+      return parser.parse(JsonUtil.mapper().readValue(json, JsonNode.class), args);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
```

Comment:
Trying to think of a better abstraction so we don't need this and the FromJsonWithArgs